modules/monitoring-setting-up-pod-topology-spread-constraints.adoc

```diff
@@ -0,0 +1,189 @@
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    prometheusK8s:
+      topologySpreadConstraints:
+      - maxSkew: 1
+        topologyKey: monitoring
+        whenUnsatisfiable: DoNotSchedule
+        labelSelector:
+          matchLabels:
+            app.kubernetes.io/name: prometheus
+----
+
+. Save the file to apply the changes automatically.
++
+[WARNING]
+====
+When you save changes to the `cluster-monitoring-config` config map, the pods and other resources in the `openshift-monitoring` project might be redeployed.
+The running monitoring processes in that project might also restart.
+====
+
+* *To configure pod topology spread constraints for user-defined monitoring:*
+endif::openshift-dedicated,openshift-rosa[]
+
+. Edit the `user-workload-monitoring-config` config map in the `openshift-user-workload-monitoring` project:
++
+[source,terminal]
+----
+$ oc -n openshift-user-workload-monitoring edit configmap user-workload-monitoring-config
+----
+
+. Add the following settings under the `data/config.yaml` field to configure pod topology spread constraints:
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: user-workload-monitoring-config
+  namespace: openshift-user-workload-monitoring
+data:
+  config.yaml: |
+    <component>: # <1>
+      topologySpreadConstraints:
+      - maxSkew: <n> # <2>
+        topologyKey: monitoring # <3>
+        whenUnsatisfiable: <value>> # <4>
+        labelSelector:
+          matchLabels: # <5>
+            <label>
+----
+<1> Specify a name of the component for which you want to set up pod topology spread constraints.
+<2> Specify a numeric value for `maxSkew`, which defines the degree to which pods are allowed to be unevenly distributed.
+This field is required, and the value must be greater than zero.
+The value specified has a different effect depending on what value you specify for `whenUnsatisfiable`.
+<3> Specify a key of node labels for `topologyKey`.
+This field is required.
+Nodes that have a label with this key and identical values are considered to be in the same topology.
+The scheduler tries to put a balanced number of pods into each domain.
+<4> Specify a value for `whenUnsatisfiable`.
+This field is required.
+Available options are `DoNotSchedule` and `ScheduleAnyway`.
+Specify `DoNotSchedule` if you want the `maxSkew` value to define the maximum difference allowed between the number of matching pods in the target topology and the global minimum.
+Specify `ScheduleAnyway` if you want the scheduler to still schedule the pod but to give higher priority to nodes that might reduce the skew.
+<5> Specify a label for `matchLabels` to identify the set of matching pods to which to apply the constraints.
++
+.Example configuration for Thanos Ruler
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: cluster-monitoring-config
+  namespace: openshift-monitoring
+data:
+  config.yaml: |
+    thanosRuler:
```

Comment:
For example thanos ruler is only in user defined-space, but prometheus are in both: 
here is an example of prometheus config in user defined space:
https://github.com/openshift/cluster-monitoring-operator/blob/7be0b5675eccd22baaf1c4f35c3f9dd0b9160fdb/pkg/manifests/manifests_test.go#L1615

Same than here:
https://github.com/marioferh/cluster-monitoring-operator/blob/7be0b5675eccd22baaf1c4f35c3f9dd0b9160fdb/pkg/manifests/manifests_test.go#L1435